sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/select/GremlinSqlSelectSingle.java

```diff
@@ -126,11 +126,11 @@ protected void runTraversalExecutor(final GraphTraversal<?, ?> graphTraversal,
             generateDataRetrieval(gremlinSqlIdentifiers, __.__());
 
             // Generate actual traversal.
+            applyHaving(graphTraversal);
+            applyWhere(graphTraversal);
```

Comment:
I don't know if this is applied bottom up or top-down, but I would expect that a where and having to be applied on different sides of the group-by.

Reply:
Yes, you are exactly correct. I actually just changed this because I noticed I broke my HAVING tests after. I will push this correction.